copy/copy.go

```diff
@@ -646,6 +688,27 @@ func (c *copier) copyOneImage(ctx context.Context, policyContext *signature.Poli
 	// If encrypted and decryption keys provided, we should try to decrypt
 	ic.diffIDsAreNeeded = ic.diffIDsAreNeeded || (isEncrypted(src) && ic.c.ociDecryptConfig != nil) || ic.c.ociEncryptConfig != nil
 
+	options.OptimizeDestinationImageAlreadyExists = true // TODO: PR: Remove forced enablement for PR testing
+	// If enabled, fetch and compare the destination's manifest. And as an optimization skip updating the destination iff equal
+	// To be enabled, options.OptimizeDestinationImageAlreadyExists must be true, the image must not be signed or encryted. And must
+	// not have any pending manifest updates
+	isSigned := len(sigs) > 0 || options.SignBy != "" // TODO: Consider allowing signatures updates only and skipping the image's layers/manifest copy if possible
+
+	logrus.Debugf("Evaluating image manifest check optimzation: options.OptimizeDestinationImageAlreadyExists=%t !isSigned=%t !destRequiresOciEncryption=%t ic.noPendingManifestUpdates()=%t", options.OptimizeDestinationImageAlreadyExists, !isSigned, !destRequiresOciEncryption, ic.noPendingManifestUpdates())
```

Comment:
Please wrap all of this in `if options.OptimizeDestinationImageAlreadyExists`, so that this log line does not appear if the caller does not opt in (and the presence of that log line is enough to indicate the value of that flag, so it doesn’t need to be logged separately); that way we don’t need to compute `isSigned` if the caller does not opt in.